pkg/asset/cluster/aws/aws.go

```diff
@@ -207,17 +234,25 @@ func tagSharedIAMProfiles(ctx context.Context, clusterID string, installConfig *
 
 	logrus.Debugf("Tagging shared instance profiles: %v", sets.List(iamProfileNames))
 
-	session, err := installConfig.AWS.Session(ctx)
+	cfg, err := configv2.LoadDefaultConfig(ctx, configv2.WithRegion(installConfig.Config.Platform.AWS.Region))
 	if err != nil {
-		return errors.Wrap(err, "could not create AWS session")
+		return fmt.Errorf("failed loading default config: %w", err)
```

Comment:
nit: Another place to change to `failed to load AWS config` while you are at it haha